diofant/sets/sets.py

```diff
@@ -949,7 +948,9 @@ def _union(self, other):
 
     @property
     def boundary(self):
-        return FiniteSet(self.start, self.end)
+        return FiniteSet(*(p for p, c in [(self.start, not self.left_open),
+                                          (self.end, not self.right_open)]
+                           if abs(p) != oo or c))
```

Comment:
> The boundary of an interval contains the endpoints regardless of the type of the interval.

That depends on the topology, right?  Are you sure, that in the standard topology on the `ℝ=(-oo, oo)` it's true?  There is no open set that contains `oo`, for example.  Also, `oo` not in the closure of the `ℝ` (it's a closed set and `oo` not it's member).

Reply:
> oo not in the closure of the ℝ

oo must be in the closure of R. Otherwise it could not be the limit of any sequences of real numbers.

The standard topology on ordered sets like R and R_bar(= [-oo, oo]) is the [order topology](https://en.wikipedia.org/wiki/Order_topology). Its subbase consists of "half-bounded" intervals like (a, oo] and [-oo, b] in R_bar. In particular, intervals like (a, oo] are open neighborhoods of oo (and always contain elements of R).

Reply:
Hmm, shouldn't be the boundary - a complement of the set interior in its closure?  Given R is open and closed - its boundary seems to be empty.

Reply:
Its boundary is also relative to the topological space. Its boundary is empty in R but nonempty in R_bar. If it is decided that `R.is_closed` should be relative to R (i.e. True), then its boundary should be empty, I guess... In that case, it looks like -oo and oo should not be boundary points of any interval in R. (But maybe there could be "extended intervals".)